### ISSUE TITLE
datepicker.js.php highlighting of selected week/month dates

### ISSUE THREAD

**Issue by William-Haller** [meeting-room-booking-system/mrbs-code#3496]
On upgrade from 1.10 to 1.11.3, the proper days weren't highlighted in the mini calendars for week or month display modes. PHP 8.1, mariadb 10.6.15, OS Debian bullseye with man backports. Apache 2.4.56. Checked with Brave 1.57.53 and Firefox 102.14/esr on Debian bookworm.

This appears to be happening due to javascript Date converting date strings to and from UTC. For example, in weekStart, the diff value is one for today (Tuesday) when it should be two. This causes the highlighting to be off by one.

Doing something similar to...

~~~js
var now = new Date;
var offset = now.getTimezoneOffset() / 60;
var tzoffset = offset.toString().padStart(2, "0");
var startstring = date + "T12:00:00.000-" + tzoffset + ":00";
var d = new Date(startstring);
~~~

instead of d = new Date(date); everywhere this happens from weekStart down, works for us, but would need a bit more robustness for time zones east of GMT. There may also be a better way to handle it completely. But as it is, when it takes the string to make the Date originally, it seems to assume it is in UTC so when it returns the value as our timezone America/Denver, it starts out a day earlier and thus messes up the diff computation. $timezone is properly set, the days to hide is set to (0,6).

**Comment by campbell-m** [comment]
OK, I'll take a look later.

**Comment by William-Haller** [comment]
A cleaner fix that would handle east/west

function datetzadjust(date) {
  var now = new Date;
  var offset = now.getTimezoneOffset() / 60;
  var tzoffset;
  if (offset > 0) {
    tzoffset = '-' + offset.toString().padStart(2, "0");
  }
  else {
    tzoffset = '+' + offset.toString().padStart(2, "0");
  }
  return date + "T12:00:00.000" + tzoffset + ":00";
}

function weekStart(date, weekStarts) {
  var d = new Date(datetzadjust(date));
  var diff = d.getDay() - weekStarts;
  if (diff < 0)
  {
   diff += 7;
  }
  d.setDate(d.getDate() - diff);

  return d.toISOString().split('T')[0];
}

As  usual, permission is granted to use the code provided for your project. Still may be issues if time zone offset not evenly divisible by 60. No rush. Thanks.

**Comment by campbell-m** [comment]
I've implemented a fix in [PR 3498](https://github.com/meeting-room-booking-system/mrbs-code/pull/3498).  Could you test it please?

**Comment by campbell-m** [comment]
Also could you double-check please the timezone for the area in question?  I know you have `$timezone` set correctly, but could you check the setting when you edit the area from the Rooms page.  Thanks.

**Comment by campbell-m** [comment]
Fixed by [PR #3498](https://github.com/meeting-room-booking-system/mrbs-code/pull/3498).
